framework/src/materials/MaterialPropertyInterface.C

```diff
@@ -88,6 +88,31 @@ MaterialPropertyInterface::defaultMaterialProperty(const std::string & name)
   return nullptr;
 }
 
+const MaterialProperty<Real> *
+MaterialPropertyInterface::zeroMaterialPropertyDot(const std::string & name)
+{
+  std::istringstream ss(name);
+  Real real_value;
+
+  // check if the string parsed cleanly into a Real number
+  if (ss >> real_value && ss.eof())
```

Comment:
My suggestion would be to factor this check out into the caller and name this method ` zeroMaterialProperty` without the `Dot`. There are applications like the derivative material system that need to generate and return zero properties. I have code that does that already, but may want to refactor my code to use this new function.

Reply:
I actually just noticed that there is a function `getZeroMaterialProperty`. I should just use that. Checking constant value will be in `getMaterialPropertyDot`.

Reply:
So @dschwen I guess `getZeroMaterialProperty` will work for you and you will not need `zeroMaterialProperty`, right?

Reply:
Given that I wrote `getZeroMaterialProperty`... yes.